pkg/view/dashboard.go

```diff
@@ -151,3 +152,123 @@ func calculateActionItemReportTrend(previous *model.ActionItemReport, current *m
 
 	return rs
 }
+
+type EngineringHealthResponse struct {
+	Data EngineeringHealthData `json:"data"`
+}
+
+type EngineeringHealthData struct {
+	Average []*EngineeringHealth    `json:"average"`
+	Groups  *GroupEngineeringHealth `json:"groups"`
+}
+
+type EngineeringHealth struct {
+	Quarter string  `json:"quarter"`
+	Value   float64 `json:"avg"`
+	Trend   float64 `json:"trend"`
+}
+
+type GroupEngineeringHealth struct {
+	Delivery      []*EngineeringHealth `json:"delivery"`
+	Quality       []*EngineeringHealth `json:"quality"`
+	Collaboration []*EngineeringHealth `json:"collaboration"`
+	Feedback      []*EngineeringHealth `json:"feedback"`
+}
+
+func ToEngineeringHealthData(average []*model.AverageEngineeringHealth, groups []*model.GroupEngineeringHealth) *EngineeringHealthData {
+	rs := &EngineeringHealthData{}
+
+	// Reverse quarter order
+	for i, j := 0, len(average)-1; i < j; i, j = i+1, j-1 {
+		average[i], average[j] = average[j], average[i]
+	}
+
+	for i, j := 0, len(groups)-1; i < j; i, j = i+1, j-1 {
+		groups[i], groups[j] = groups[j], groups[i]
+	}
+
+	for _, a := range average {
+		rs.Average = append(rs.Average, &EngineeringHealth{
+			Quarter: strings.Split(a.Quarter, "/")[1] + "/" + strings.Split(a.Quarter, "/")[0],
+			Value:   a.Avg,
+		})
+	}
+
+	calculateTrendForEngineeringHealthList(rs.Average)
+
+	rs.Groups = toGroupEngineeringHealth(groups)
+
+	calculateTrendForEngineeringHealthList(rs.Groups.Delivery)
+	calculateTrendForEngineeringHealthList(rs.Groups.Collaboration)
+	calculateTrendForEngineeringHealthList(rs.Groups.Quality)
+	calculateTrendForEngineeringHealthList(rs.Groups.Feedback)
+
+	return rs
+}
+
+func toGroupEngineeringHealth(groups []*model.GroupEngineeringHealth) *GroupEngineeringHealth {
+	rs := &GroupEngineeringHealth{}
+	count := 0
+	quarter := ""
+
+	for _, g := range groups {
+		if quarter != g.Quarter {
+			count++
+			quarter = g.Quarter
+
+			if count > 4 {
+				break
+			}
+		}
+
+		switch g.Area {
+		case "Delivery performance":
+			if len(rs.Delivery) < 4 {
```

Comment:
Why we check length smaller than 4 ?

Reply:
To make sure have less than 4 quarter, but I think the condition
```
if count > 4 {
				break
			}
```
is enough, thank you